docs/DeveloperGuide.adoc

```diff
@@ -763,20 +848,197 @@ Use case ends.
 * 3a. The given index is invalid.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1. iContacts shows an error message.
 +
 Use case resumes at step 2.
 
-{More to be added}
+[discrete]
+=== Use case: UC2 -  Export Contacts
+
+*MSS*
+
+1.  User selects contacts to be exported based on current list
+2.  User requests for creating file containing selected contacts
+3.  iContacts creates file with the link:#contact-information[contact information] of specified contacts
+4.  User specifies the location where link:#data-file[data file] needs to be stored
+5.  iContacts saves the data file in proper format and indicates success
+6.  User sends the data file to the required recipient
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. No contacts selected
++
+Use case ends.
+
+* 3a. iContacts unable to create file with mentioned contacts
++
+[none]
+** 3a1. iContacts shows an error message.
++
+Use case resumes at step 1.
+[none]
+* 5a. iContacts unable to store at given location
+** 5a1. iContacts shows an error message.
++
+Use case ends
+* *a. At any time user chooses to cancel exporting contacts
++
+[none]
+** *a1. iContacts shows an error message.
++
+Use case ends.
+
+
+[discrete]
+=== Use case: UC3 - Import Contacts
+
+*Precondition*
+
+[none]
+*  User receives the link:#data-file[data file] containing contacts
+
+
+*MSS*
+
+
+1.  User requests to import contacts
+2.  iContacts requests for file location
+3.  User specifies location where the data file is found
+4.  iContacts reads the file and merges the contacts from the received data file into the data file that iContacts is currently using
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. No file selected
++
+Use case ends.
+
+* 4a. iContacts finds duplicate contacts
++
+[none]
+** 4a1. iContacts shows an error message and doesn't merge the duplicate ones
```

Comment:
does iContacts stop merging completely? or only duplicates are not merged? can be specific here

Reply:
Only duplicate ones are not merged. Made the necessary change